src/bluetooth.cpp

```diff
@@ -1712,13 +1555,19 @@ void bluetooth::deviceDiscovered(const QBluetoothDeviceInfo &device) {
                 chronoBike->deviceDiscovered(b);
                 // NOTE: Commented due to #358
                 // connect(this, SIGNAL(searchingStop()), chronoBike, SLOT(searchingStop()));
+            }
+
+            if(newDevice) {
+                this->bluetoothDevice = newDevice;
```

Comment:
If bluetoothDevice is already set, this is a memory leak. Previously the object for the found device would be stored in the dedicated field then deleted later. This needs more thought.